tools/YamlSplitter/FragmentModelHelper.cs

```diff
@@ -0,0 +1,192 @@
+    using Microsoft.DocAsCode.MarkdigEngine.Extensions;
+    using Microsoft.DocAsCode.Plugins;
+    using Microsoft.DocAsCode.Tools.YamlSplitter.Models;
+
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public static class FragmentModelHelper
+    {
+        public static MarkdigMarkdownService MDService = new MarkdigMarkdownService(
+            new MarkdownServiceParameters
+            {
+                BasePath = ".",
+                Extensions = new Dictionary<string, object>
+                    {
+                        { LineNumberExtension.EnableSourceInfo, false }
+                    }
+            });
+
+        public static Dictionary<string, FragmentFile> LoadMarkdownFolder(string path)
+        {
+            ConcurrentBag<FragmentFile> fileBag = new ConcurrentBag<FragmentFile>();
+            int errorHappened = 0;
+            Parallel.ForEach(Directory.EnumerateFiles(path, "*.md", SearchOption.AllDirectories), filePath =>
+            {
+                try
+                {
+                    var fragments = LoadMarkdownFragment(filePath);
+                    if (fragments != null)
+                    {
+                        fileBag.Add(new FragmentFile()
+                        {
+                            FilePath = filePath,
+                            FragmentsByUid = fragments
+                        });
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine("Error loading " + filePath + ": " + ex.ToString());
+                    Interlocked.Increment(ref errorHappened);
+                }
+            });
+            if (errorHappened > 0)
+            {
+                Console.WriteLine("Error loading {0} files, aborting...", errorHappened);
+                return null;
+            }
+            errorHappened = 0;
+            var filesByUid = new Dictionary<string, FragmentFile>();
+            foreach (var fragmentFile in fileBag)
+            {
+                foreach (var fragmentPair in fragmentFile.FragmentsByUid)
+                {
+                    if (filesByUid.ContainsKey(fragmentPair.Key))
+                    {
+                        Console.WriteLine("Dup Uid found in below two files:");
+                        Console.WriteLine(fragmentFile.FilePath);
+                        Console.WriteLine(filesByUid[fragmentPair.Key].FilePath);
+                        errorHappened++;
+                    }
+                    else
+                    {
+                        filesByUid.Add(fragmentPair.Key, fragmentFile);
+                    }
+                }
+            }
+            if (errorHappened > 0)
+            {
+                Console.WriteLine("Dup Uid found in {0} files, aborting...", errorHappened);
```

Comment:
question: for different groups, uid can be duplication, do we need run yamlsplitter for each group?

Reply:
this function is no longer needed, so I deleted it